x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/index/IndexResolver.java

```diff
@@ -222,64 +234,151 @@ private void filterResults(String javaRegex, GetAliasesResponse aliases, GetInde
+
+        int dot = fieldName.lastIndexOf('.');
+        String fullFieldName = fieldName;
+
+        if (dot >= 0) {
+            String parentName = fieldName.substring(0, dot);
+            fieldName = fieldName.substring(dot + 1);
+            EsField parent = flattedMapping.get(parentName);
+            if (parent == null) {
+                Map<String, FieldCapabilities> map = globalCaps.get(parentName);
+                FieldCapabilities parentCap = map == null ? null : map.values().iterator().next();
+                parent = createField(parentName, parentCap, globalCaps, hierarchicalMapping, flattedMapping, true);
             }
+            parentProps = parent.getProperties();
         }
-        if (merged != null) {
-            // at this point, we are sure there's the same mapping across all (if that's the case) indices
-            // to keep things simple, use the given pattern as index name
-            merged = IndexResolution.valid(new EsIndex(indexWildcard, merged.get().mapping()));
-        } else {
-            merged = IndexResolution.notFound(indexWildcard);
+
+        EsField field = null;
+        Map<String, EsField> props = hasChildren ? new TreeMap<>() : emptyMap();
+
+        DataType esType = DataType.fromEsType(caps.getType());
+        switch (esType) {
+            case TEXT:
+                field = new TextEsField(fieldName, props, false);
+                break;
+            case KEYWORD:
+                int length = DataType.KEYWORD.defaultPrecision;
+                // to check whether isSearchable/isAggregateable takes into account the presence of the normalizer
+                boolean normalized = false;
+                field = new KeywordEsField(fieldName, props, caps.isAggregatable(), length, normalized);
+                break;
+            case DATE:
+                field = new DateEsField(fieldName, props, caps.isAggregatable());
+                break;
+            case UNSUPPORTED:
+                field = new UnsupportedEsField(fieldName, caps.getType());
+                break;
+            default:
+                field = new EsField(fieldName, esType, props, caps.isAggregatable());
         }
-        return merged;
+
+        parentProps.put(fieldName, field);
+        flattedMapping.put(fullFieldName, field);
+
+        return field;
+    }
+    
+    private static FieldCapabilitiesRequest createFieldCapsRequest(String index) {
+        return new FieldCapabilitiesRequest()
+                .indices(Strings.commaDelimitedListToStringArray(index))
+                .fields("*")
+                //lenient because we throw our own errors looking at the response e.g. if something was not resolved
+                //also because this way security doesn't throw authorization exceptions but rather honors ignore_unavailable
+                .indicesOptions(IndicesOptions.lenientExpandOpen());
     }
 
+    // Concrete indices still uses get mapping
```

Comment:
maybe add a TODO to be more clear?

Reply:
it's rather WIP blocked by : https://github.com/elastic/elasticsearch/pull/34071